### PR TITLE
WRN-3837: [Moonstone] Migrate Panels:Header, IndexedBreadcrumb and Panels unit tests to Testing Library

### PR DESCRIPTION
**Pull request by adrian-cocoara-lgp**
Enact-DCO-1.0-Signed-off-by: Adrian Cocoara adrian.cocoara@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Migrated `Header`, `IndexedBreadcrumb` and `Panels` to react-testing-library  

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Before migrating to react-testing-library, all tests for `Panels were skipped because at the time they were created enzyme couldn't handle some tests. After migrating to react-testing-library all tests pass, so none of them are skipped any more

### Links
[//]: # (Related issues, references)
WRN-3837

### CODE REVIEW

File: Panels/tests/Panels-specs.js [hunk 1]
```diff
@@ -1,140 +1,106 @@
...
+		fireEvent.click(applicationCloseButton);
+
+		expect(handleAppClose).toHaveBeenCalled();
+	});
 
-			const expected = 'last-focused';
-			const actual = panels.find('Panel').prop('autoFocus');
+	test('should set application close button "aria-label" to closeButtonAriaLabel', () => {
+		const label = 'custom close button label';
+		render(
```
Comment (alexandrumorariu) [inline Panels/tests/Panels-specs.js]: Please write this render method in a single line.

File: Panels/tests/IndexedBreadcrumbs-specs.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
-import {mount} from 'enzyme';
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
 import IndexedBreadcrumbs from '../IndexedBreadcrumbs';
 
 describe('IndexedBreadcrumbs', () => {
```
Comment (alexandrumorariu) [inline Panels/tests/IndexedBreadcrumbs-specs.js:6]: Please remove the empty line between describe and the comment

File: Panels/tests/Panels-specs.js [hunk 3]
```diff
@@ -1,140 +1,106 @@
...
+
+		expect(actual).toBe(expected);
+	});
+
+	test('should not set {autoFocus} on child when navigating to a higher index when it has an autoFocus prop set', () => {
+		const DivPanel = ({autoFocus, id}) => <div data-testid="panel" id={id}>{autoFocus}</div>;
+		const {rerender} = render(
+			<Panels index={0}>
+				<DivPanel />
+				<DivPanel id="p2" autoFocus="last-focused" />
```
Comment (alexandrumorariu) [inline Panels/tests/Panels-specs.js]: Order the props in alphabetical order(same on the rerender).

**Review by alexandrumorariu — DISMISSED** [review]
LGTM

**Review by alexandrumorariu — APPROVED** [review]
LGTM

File: Panels/tests/Header-specs.js [hunk 4]
```diff
@@ -1,42 +1,39 @@
...
 import Header from '../Header';
-import css from '../Header.module.less';
 
 describe('Header Specs', () => {
-
 	test('should render with title text without changing case', () => {
 		let msg = 'cRaZy-cased super Header';
-
-		const header = mount(
+		render(
```
Comment (stanca-pop-lgp) [inline Panels/tests/Header-specs.js]: You can put the render on a single line. Same below. The last test is ok with multiple lines.

File: Panels/tests/IndexedBreadcrumbs-specs.js [hunk 5]
```diff
@@ -37,23 +38,17 @@ describe('IndexedBreadcrumbs', () => {
...
+		);
 
-			const expected = 1;
-			const actual = handleClick.mock.calls.length;
-
-			expect(actual).toBe(expected);
-		}
-	);
+		const breadcrumb = screen.getByLabelText('GO TO PREVIOUS');
+		userEvent.click(breadcrumb);
```
Comment (stanca-pop-lgp) [inline Panels/tests/IndexedBreadcrumbs-specs.js:50]:
```suggestion
		render(<nav>{IndexedBreadcrumbs('id', 1, 1, handleClick)}</nav>);
		const breadcrumb = screen.getByLabelText('GO TO PREVIOUS');
		
		userEvent.click(breadcrumb);
```

Comment (stanca-pop-lgp) [inline Panels/tests/IndexedBreadcrumbs-specs.js]:
```suggestion
		render(<nav>{IndexedBreadcrumbs('id', 1, 1, handleClick)}</nav>);
		const breadcrumb = screen.getByLabelText('GO TO PREVIOUS');
		
		userEvent.click(breadcrumb);
```

**Review by stanca-pop-lgp — APPROVED** [review]
LGTM

**Pull request merged**